File: src/main/java/com/JamieSatchell/ims/domain/Category.java
package com.JamieSatchell.ims.domain;

import jakarta.persistence.*;
import lombok.NoArgsConstructor;
import lombok.Getter;
import lombok.Setter;
import java.util.List;

@Entity
@Getter @Setter
@NoArgsConstructor // Generates a no-arg constructor
public class Category {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;
    private String name;
    @OneToMany(mappedBy = "category")
    private List<Product> products;
}

File: src/main/java/com/JamieSatchell/ims/domain/Category.kt
package com.JamieSatchell.ims.domain

import jakarta.persistence.Entity
import jakarta.persistence.GeneratedValue
import jakarta.persistence.GenerationType
import jakarta.persistence.Id
import jakarta.persistence.OneToMany



import lombok.NoArgsConstructor

@Entity
@NoArgsConstructor
data class Category (
        @Id @GeneratedValue(strategy = GenerationType.IDENTITY)
        var id: Long? = null,
        var name: String? = null,
        @OneToMany (mappedBy = "catagory")
        var products: MutableList<Product> = mutableListOf()

)


